### PR TITLE
MON-1749: Allow users to disable the local Alertmanager

### PR DESCRIPTION
**Pull request by fpetkovski**
The ACM would like to funnel all alerts from a fleet of OpenShift
clusters through a single external Alertmanager. As a result, they
would like to trim down the monitoring stack and disable the local
alertmanager.

This commit exposes a binary field in the cluster-monitoring-configmap
which allows users to fully disable the built-in alertmanager.

* [x] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

**Review by prashbnair — APPROVED** [review]

**Comment by prashbnair** [comment]
lgtm

**Review by PhilipGough — APPROVED** [review]
/lgtm

File: test/e2e/framework/framework.go
Comment (simonpasquier) [inline test/e2e/framework/framework.go:86]: we already create `openshiftRouteClient` just below, why not use it?

**Review by simonpasquier — COMMENTED** [review]
/hold
just one small comment, I think we want to see the impact on the console too before merging.

File: test/e2e/framework/framework.go
Comment (fpetkovski) [reply test/e2e/framework/framework.go]: Good point, addressed.

**Comment by simonpasquier** [comment]
/lgtm cancel

testing locally CMO goes degraded when Alertmanager is disabled

```
Failed to rollout the stack. Error: running task Updating configuration sharing failed: failed to retrieve Alertmanager host: getting Route object failed: routes.route.openshift.io "alertmanager-main" not found
```

**Comment by fpetkovski** [comment]
Ack, do we still need that public URL though? Can we remove it from the shared config?

**Comment by simonpasquier** [comment]
@PhilipGough has https://github.com/openshift/cluster-monitoring-operator/pull/1223 in progress to remove the shared configmap but it's still waiting on code changes in the console repository.

**Comment by fpetkovski** [comment]
@simonpasquier we had a chat with @kyoto and his assessment is that it's fine to merge this PR before the console changes are in place. The reason for this once the AM is disabled, the console shows a message that it cannot connect to it and there isn't a major degradation to the UI. 

Also, @kyoto will speak to the PM during planning to see how much bandwidth he has for the next sprint. If he has no time to handle the console changes, I will take over the work and he will review the PR.

**Comment by simonpasquier** [comment]
Thanks for the heads-up, I agree with @kyoto that the current UX in the console is good enough (disabling Alertmanager is a deliberate choice so you shouldn't be too surprised that the console fails to find the local Alertmanager).

**Comment by simonpasquier** [comment]
I don't see any test in `ConfigSharingTask` that verifies whether the local Alertmanager is disabled (just like we do for Grafana)? Am I missing something?

https://github.com/openshift/cluster-monitoring-operator/blob/e3023648dee4860e07b04e329f2a64c4f50061ef/pkg/tasks/configsharing.go#L63-L73

**Comment by fpetkovski** [comment]
Based on one of your previous comments, I thought we would wait for @PhilipGough's task to remove the shared config to be merged. Let me know if I should handle the case in this PR instead.

**Comment by simonpasquier** [comment]
Sorry for the confusion :) Yes let's handle it in this PR because we're not sure when the other one is going to be merged.

**Comment by fpetkovski** [comment]
@simonpasquier to make the review easier, I added a new commit which handles this edge case and removes the public url from the shared configmap when AM is disabled.

**Review by simonpasquier — COMMENTED** [review]
/lgtm

File: pkg/manifests/manifests.go
Comment (simonpasquier) [inline pkg/manifests/manifests.go:1065]:
```suggestion
	if amConfigs == nil {
		if !alertmanagerEnabled {
			s.StringData["alertmanagers.yaml"] = `alertmanagers: []`
		}
		return s, nil
	}
```

File: test/e2e/alertmanager_test.go
Comment (simonpasquier) [inline test/e2e/alertmanager_test.go:380]:
```suggestion
		{name: "assert role monitoring-alertmanager-edit does not exist", assertion: f.AssertRoleDoesNotExist("monitoring-alertmanager-edit", "openshift-monitoring")},
```

**Comment by dgrisonnet** [comment]
/retitle MON-1749: Allow users to disable the local Alertmanager

**Comment by fpetkovski** [comment]
@juzhao if this is good to merge, could you add the qe approval label?

@jan--f could you ping the px and docs representatives for our team?

**Comment by sferich888** [comment]
Would users have to turn off the local 'alertmanager' for the information to be aggregated to ACM? What if I wanted to get alerts on cluster; and within ACM? Would that be possible? 


cc: @jcarter12

**Comment by fpetkovski** [comment]
@sferich888 users can turn the local alertmanager and configure external alertmanagers as independent features. So you they can do both depending on how they configure CMO.

File: pkg/manifests/manifests.go
Comment (raptorsun) [inline pkg/manifests/manifests.go:1075]: just curious, why we care additional config for alter manager while it is disabled? 🤔

**Pull request merged**
